bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/common/PriceImpl.java

```diff
@@ -197,4 +213,32 @@ public Double getDiscountPercentMax() {
         return isRange() ? discountPercentMax : Double.NaN;
     }
 
-}
+    @Override
```

Comment:
Do we really need to extend the `Price` interface here? 

Afaics we the open amount is a price range as well. Can we reuse the `finalPriceMin`/`finalPriceMax` values here instead? 

1) `finalPriceMin` and `finalPriceMax` set, price range "From XY to YZ"
2) `finalPriceMin` set and `finalPriceMax` null, price range "From XY" 

`minStartingPrice` and `isMinStartingPrice` are unnecessary. 

See more details in the comments to `price.html`

Reply:
`open_amount_min` and  `open_amount_max` is not the part of price_range in Graphql . `finalPriceMin` and  `finalPriceMax` only reflects the min of `giftcard_amounts`  which is fixed list in GiftCard.We have extended Price class to expose the open_amount_min and open_amount_max on UI.

Reply:
Thanks for your explanation, this is exactly my point. The `Price` interface is used by the htl frontend. It does not need about the details of graphql and the backend. Putting that aside the `openAmountMin` and `openAmountMax` are the lower and upper bound of a price range and we should reuse the fields we have for that already, extending the frontend to handle an undefined upper end as explained for the `price.html`.

Reply:
`isMinStartingPrice `  is required  on UI  to identify  the GiftCard product type on category page and  search page to show the minimum price for  GiftCard.

Reply:
As said, it may not be required as discussed above. You either have a min and max price in the range which should result in a rendering of `From XY to YZ` (as is) or you only have a min price which should result in a rendering of `From XY` (new).

Reply:
How to cater the use case when we have fixed list and open amount both enabled for the GiftCardProduct.
eg:  Fixed list  amount  -> $5,$20,$30
        OpenAmountMin    -> $10
        OpenAmountMax   ->  $20
        For Gift  Card now we need 3 values-> $5 for category page to show  during listing and OpenAmountMin ,OpenAmountMax for PDP page.If We reuse  the `finalPriceMin` and equate its value with OpenAmountMin the value  shown on the category page would be $10 but we need to show $5 and on PDP page , it needs to be $10

Reply:
On the productlist component (category page) you should list the giftcard with 'From $5 to $30'. Showing only one end of the price range when both are given should either be a style or policy option of the component, but it is not related to gift cards. 

I don't understand what the use case of having a smaller "open amount max" then "fixed max". From my pov if you allow a fixed 10,20,30, open amount should go at least a high a 30 or even higher. Could you go more in detail here?

Reply:
Yes,It should   be higher in actual product configuration but from the functionality perspective, currently this is how the Magento is showing  up.If we have both -fixed list as well as OpenAmountMin and OpenAmountMax  defined for product then on category page it Final Price Min, and for PDP page we have openAmountMin and openAmountMax mentioned.
Here the price shown is final price min which $5 
<img width="1390" alt="Screenshot 2021-07-19 at 11 53 30 PM" src="https://user-images.githubusercontent.com/22527264/126208408-db1a398e-4ded-4448-9a85-4b5d4b59129f.png">
and openAmountMin and Max for the PDP page
<img width="1466" alt="Screenshot 2021-07-19 at 11 54 21 PM" src="https://user-images.githubusercontent.com/22527264/126208617-01e63e6f-b64e-4aa4-a9dc-1b02240a721d.png">